src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,103 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] values;
+    private int size;
+
+    public ArrayList(int initCapacity) {
+        if (initCapacity <= 0) {
+            throw new IllegalArgumentException();
+        }
+        values = (T[]) new Object[initCapacity];
+    }
+
+    public ArrayList() {
+        this(DEFAULT_CAPACITY);
+    }
```

Comment:
it's overcomplicated
since DEFAULT_CAPACITY is constant you can create only one conctructor
```suggestion
    public ArrayList() {
        values = (T[]) new Object[DEFAULT_CAPACITY];
    }
```